browser/brave_rewards/android/brave_rewards_native_worker.cc

```diff
@@ -109,6 +112,30 @@ bool BraveRewardsNativeWorker::IsRewardsEnabled(JNIEnv* env) {
       ->GetBoolean(brave_rewards::prefs::kEnabled);
 }
 
+bool BraveRewardsNativeWorker::ShouldShowSelfCustodyInvite(JNIEnv* env) {
+  if (base::FeatureList::IsEnabled(
+          brave_rewards::features::kAllowSelfCustodyProvidersFeature)) {
+    auto& self_custody_dict =
+        ProfileManager::GetActiveUserProfile()
+            ->GetOriginalProfile()
+            ->GetPrefs()
+            ->GetDict(brave_rewards::prefs::kSelfCustodyAvailable);
+
+    bool isSelfCustodyInviteDismissed =
+        ProfileManager::GetActiveUserProfile()
+            ->GetOriginalProfile()
+            ->GetPrefs()
+            ->GetBoolean(brave_rewards::prefs::kSelfCustodyInviteDismissed);
+
+    if (auto solana_entry = self_custody_dict.FindBool(
+            brave_rewards::internal::constant::kWalletSolana);
+        solana_entry && *solana_entry && !isSelfCustodyInviteDismissed) {
+      return true;
```

Comment:
Note that this function doesn't currently take into account whether or not "solana" is allowed for the user's region (as returned by the `/parameters` endpoint).